cwltool/docker.py

```diff
@@ -341,9 +341,11 @@ def create_runtime(
                 runtime = [user_space_docker_cmd, "run"]
         else:
             runtime = ["docker", "run", "-i"]
+        os.chmod(os.path.realpath(self.outdir), 0o777)
```

Comment:
Does it have to be world writeable?

https://github.com/common-workflow-language/cwltool/pull/1516/checks?check_run_id=3568589880#step:9:24

Reply:
yes, this what we had to.

While running, I get: 

```
(base) [jovyan@jupyter-fbrito-dev workspace]$ ll -a /tmp/eatw95id
total 65932
drwxrwxrwx 2 jovyan users      4096 Sep 14 10:41 .
drwxrwxrwt 1 root   root       4096 Sep 14 10:39 ..
-rw-r--r-- 1 101000 100099 67505506 Sep 14 10:41 S2B_53HPA_20210723_0_L2A.tif
```

while running: 

```
INFO [job band_math.cwl] /tmp/eatw95id$ docker \
    run \
    -i \
    --mount=type=bind,source=/tmp/eatw95id,target=/BixLsD \
    --mount=type=bind,source=/tmp/bnll6nbs,target=/tmp \
    --workdir=/BixLsD \
    --read-only=true \
    --user=1001:100 \
    --rm \
    --cidfile=/tmp/catzbo8t/20210914103914-031344.cid \
    --env=TMPDIR=/tmp \
    --env=HOME=/BixLsD \
    docker.io/terradue/otb-7.2.0 \
    otbcli_BandMathX \
    -out \
    S2B_53HPA_20210723_0_L2A.tif \
    -exp \
    '(im3b1 == 8 or im3b1 == 9 or im3b1 == 0 or im3b1 == 1 or im3b1 == 2 or im3b1 == 10 or im3b1 == 11) ? -2 : (im1b1 - im2b1) / (im1b1 + im2b1)' \
    -il \
    /vsicurl/https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/53/H/PA/2021/7/S2B_53HPA_20210723_0_L2A/B8A.tif \
    /vsicurl/https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/53/H/PA/2021/7/S2B_53HPA_20210723_0_L2A/B12.tif \
    /vsicurl/https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/53/H/PA/2021/7/S2B_53HPA_20210723_0_L2A/SCL.tif
```

with: 

```
(base) [jovyan@jupyter-fbrito-dev workspace]$ id
uid=1001(jovyan) gid=100(users) groups=100(users)
```

So anything below 777, fails with a permission denied